### PR TITLE
ENTSWM-256: BOM in the repo zip, option to enable repo zip creation w…

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
…ith license mojo

### CODE REVIEW

**Comment by mkouba** [comment]
Could you describe the changes a little bit? ;-)

**Comment by michalszynkiewicz** [comment]
There are two things done in this PR:
1. The BOM is installed into the repository (https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/63/files#diff-f994d27d3d7c3cd5667a30e602cc05ccR153)
Before the changes, the BOM that lists the productized fractions was not available in the repository
2. Maven repo zip generation with RepositoryBuilderMojo is now possible within license mojo, it is skipped by default, but if someone sets `<generateZip>true</generateZip>` in the plugin configuration, the zip will get generated (https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/63/files#diff-f994d27d3d7c3cd5667a30e602cc05ccR77 and https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/63/files#diff-f994d27d3d7c3cd5667a30e602cc05ccR277

Moreover, a `maven-repository` classifier has been added to the maven repo zip.

File: src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java [hunk 1]
```diff
@@ -142,6 +150,33 @@ private void executeGeneratedProjectBuild(File pomFile, File projectDir, File re
...
+                project.getArtifactId(), project.getVersion(), "pom"
+        );
+
+        File targetFile = new File(targetDir, fileName);
+
+        FileUtils.copyFile(getBomFile(), targetFile);
+        getLog().info("Copied the bom to " + targetFile.getAbsolutePath());
+    }
+
+    private File createSubDirectory(File repoDir, String... subPath) {
```
Comment (mkouba) [inline src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java]: Hm, this method actually does not create any directory. It only creates a `File` instance and could be probably replaced with `java.nio.file.FileSystem.getPath(String, String...).toFile()`?

Comment (michalszynkiewicz) [reply src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java]: yes, copyFile creates the required directories. I'll try replacing `createSubDirectory` with `getPath`.

Comment (michalszynkiewicz) [reply src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java]: Done

**Review by mkouba — APPROVED** [review]
Looks good.

**Pull request merged**